File: xamarin/SurgeXamarinAndroid/SurgeXamarinAndroidBindings/Additions/PlayerCallback.cs
using System;
using System.Runtime.InteropServices;

public class PlayerCallback: Java.Lang.Object, CO.Instil.Surge.Callbacks.IPlayerCallback {

    private Action<CO.Instil.Surge.Player.RtspErrorCode> callback;

    public PlayerCallback(Action<CO.Instil.Surge.Player.RtspErrorCode> callback) {
        this.callback = callback;
    }

    public void Response(CO.Instil.Surge.Player.RtspErrorCode errorCode) {
        if (callback != null) {
            callback(errorCode);
        }
    }
}

File: android/surge/src/main/java/co/instil/surge/callbacks/PlayerCallback.java
/*
 * Copyright (c) 2017 Instil Software.
 *
 * This file is subject to the terms and conditions defined in
 * file 'LICENSE.txt', which is part of this source code package.
 */

package co.instil.surge.callbacks;

import co.instil.surge.player.RtspErrorCode;

/**
 * Java callback interface to allow a user to define a callback that provides an RTSP error code as a response.
 */
public interface PlayerCallback {
    void response(RtspErrorCode errorCode);
}
